### DISCUSSION TITLE
How is Noun art encoded?

### DISCUSSION THREAD

**Discussion by cdt-eth** [nounsDAO/nouns-monorepo#395] [Solidity / smart contracts] [answered]
How is Noun art encoded?

**Comment by solimander** [comment, marked as answer]
Noun parts are run-length encoded, which means repeating, same-color pixels in each row are grouped. The exact encoding for each noun part is: `palette_index, bounds (top, right, bottom, left), [pixel_length, color_index][]`

The bounds are the bounding box of the actual content within the 32x32 pixel grid. For more information, take a look the the [on-chain art documentation](https://nouns.notion.site/Noun-Protocol-32e4f0bf74fe433e927e2ea35e52a507#6bbc1162af90427bad2dda63a3b31254).

You can use the [@nouns/sdk](https://github.com/nounsDAO/nouns-monorepo/tree/master/packages/nouns-sdk#images) to run-length encode PNG images.
